homeassistant/components/flo/sensor.py

```diff
@@ -0,0 +1,213 @@
+        super().__init__(
+            f"{device.mac_address}_daily_consumption", NAME_DAILY_USAGE, device
+        )
+        self._state: float = None
+
+    @property
+    def icon(self) -> str:
+        """Return the daily usage icon."""
+        return WATER_ICON
+
+    @property
+    def state(self) -> Optional[float]:
+        """Return the current daily usage."""
+        if self._device.consumption_today is None:
+            return None
+        return round(self._device.consumption_today, 1)
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return gallons as the unit measurement for water."""
+        return VOLUME_GALLONS
+
+    async def async_added_to_hass(self):
+        """When entity is added to hass."""
+        self.async_on_remove(
+            async_dispatcher_connect(
+                self.hass,
+                f"{self._device.mac_address}-{SIGNAL_WATER_CONSUMPTION_UPDATED}",
+                self.async_write_ha_state,
+            )
+        )
+
+
+class FloSystemModeSensor(FloEntity):
+    """Monitors the current Flo system mode."""
+
+    def __init__(self, device):
+        """Initialize the system mode sensor."""
+        super().__init__(
+            f"{device.mac_address}_current_system_mode",
+            NAME_CURRENT_SYSTEM_MODE,
+            device,
+        )
+        self._state: str = None
+
+    @property
+    def state(self) -> Optional[str]:
+        """Return the current system mode."""
+        if not self._device.current_system_mode:
+            return STATE_UNKNOWN
+        return self._device.current_system_mode
+
+
+class FloCurrentFlowRateSensor(FloEntity):
+    """Monitors the current water flow rate."""
+
+    def __init__(self, device):
+        """Initialize the flow rate sensor."""
+        super().__init__(
+            f"{device.mac_address}_current_flow_rate", NAME_FLOW_RATE, device
+        )
+        self._state: float = None
+
+    @property
+    def icon(self) -> str:
+        """Return the daily usage icon."""
+        return GAUGE_ICON
+
+    @property
+    def state(self) -> Optional[float]:
+        """Return the current flow rate."""
+        if self._device.current_flow_rate is None:
+            return None
+        return round(self._device.current_flow_rate, 1)
+
+    @property
+    def unit_of_measurement(self) -> str:
+        """Return gallons as the unit measurement for water."""
+        return "gpm"
+
+
+class FloTemperatureSensor(FloEntity):
+    """Monitors the temperature."""
+
+    def __init__(self, device):
+        """Initialize the temperature sensor."""
+        super().__init__(f"{device.mac_address}_temperature", NAME_TEMPERATURE, device)
+        self._state: float = None
+
+    @property
+    def state(self) -> Optional[float]:
+        """Return the current temperature."""
+        if self._device.temperature is None:
+            return None
+        return round(fahrenheit_to_celsius(self._device.temperature), 1)
```

Comment:
If we always convert it to celsius, better to just return it as fahrenheit with unit fahrenheit. That will cause less conversion for Fahrenheit users.

Reply:
I may have to look at this a bit more to make sure it isn't coming back in the user configured unit from the API

Reply:
that would be even better.